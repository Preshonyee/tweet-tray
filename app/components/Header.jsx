import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'styled-components';
import Theme from 'styled-theming';

import * as constants from '../constants';

const HeaderStyle = Styled.header`
    user-select: none;
    position: absolute;
    z-index: 99;
    left: 0px;
    top: 0px;
    width: 100%;
    height: 34px;
    padding-top: 8px;
    padding-bottom: 7px;
    background-color: ${Theme('mode', { day: constants.WHITE, night: constants.DARK_MODE_FOREGROUND, })};
    box-shadow: 0px 1px 6px rgba(0, 0, 0, 0.25);
`;

const LeftViewStyle = Styled.div`
    width: auto;
    height: 100%;
    float: left;
    padding-left: ${constants.SPACING}px;
    & > * {
      float: left;
    }
`;

const TitleStyle = Styled.div`
    user-select: none !important;
    font-size:  ${constants.LARGE_FONT_SIZE}px;
    font-weight: bold;
    color: ${Theme('mode', { day: constants.BLACK, night: constants.WHITE, })};
    line-height: 33px;
    float: left;
    padding-left: ${constants.SPACING}px;
    width: auto;
    height: 100%;
`;

const RightViewStyle = Styled.div`
    width: auto;
    height: 100%;
    float: right;
    padding-right: ${constants.SPACING}px;

    & > * {
      float: right;
    }
`;

const Header = (props) => {
  const { leftView, title, rightView, } = props;

  return (
    <HeaderStyle>
      {leftView !== null && (
        <LeftViewStyle>
          {leftView}
        </LeftViewStyle>
      )}
      <TitleStyle>
        {title}
      </TitleStyle>
      {rightView !== null && (
        <RightViewStyle>
          {rightView}
        </RightViewStyle>
      )}
    </HeaderStyle>
  );
};

Header.propTypes = {
  leftView: PropTypes.object,
  title: PropTypes.string.isRequired,
  rightView: PropTypes.object,
};
Header.defaultProps = {
  leftView: null,
  rightView: null,
};

export default Header;

