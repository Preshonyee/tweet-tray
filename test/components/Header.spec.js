import React from 'react';
import Enzyme, { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../../app/components/Header';

Enzyme.configure({ adapter: new Adapter(), });

function setup() {
  const component = shallow(<Header
    leftView={<div>Left View</div>}
    title="Header Title"
    rightView={<div>Right View</div>}
  />);
  return {
    component,
    leftView: component.find('Header__LeftViewStyle'),
    title: component.find('Header__TitleStyle'),
    rightView: component.find('Header__RightViewStyle'),
  };
}

describe('Header component', () => {
  it('should render', () => {
    const { component, } = setup();
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render with leftView', () => {
    const { leftView, } = setup();
    expect(shallowToJson(leftView)).toMatchSnapshot();
  });

  it('should render with title', () => {
    const { title, } = setup();
    expect(shallowToJson(title)).toMatchSnapshot();
  });

  it('should render with rightView', () => {
    const { rightView, } = setup();
    expect(shallowToJson(rightView)).toMatchSnapshot();
  });

});
