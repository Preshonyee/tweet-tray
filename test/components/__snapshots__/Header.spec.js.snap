// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
<Header__HeaderStyle>
  <Header__LeftViewStyle>
    <div>
      Left View
    </div>
  </Header__LeftViewStyle>
  <Header__TitleStyle>
    Header Title
  </Header__TitleStyle>
  <Header__RightViewStyle>
    <div>
      Right View
    </div>
  </Header__RightViewStyle>
</Header__HeaderStyle>
`;

exports[`Header component should render with leftView content 1`] = `
<div>
  Left View
</div>
`;

exports[`Header component should render with rightView content 1`] = `
<div>
  Right View
</div>
`;

exports[`Header component should render with title content 1`] = `"<Header__TitleStyle />"`;
